hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -636,61 +654,168 @@ private void insertTokenAccountRevokeKyc(RecordItem recordItem) {
+                // negative amount, later in the pipeline the token total supply will be reduced accordingly
+                Token token = Token.of(entityTokenId);
+                token.setModifiedTimestamp(consensusTimestamp);
+                token.setTotalSupply(accountAmount.getAmount());
+                entityListener.onToken(token);
+            }
+        }
+    }
+
+    private void insertTokenTransfers(RecordItem recordItem) {
+        if (!entityProperties.getPersist().isTokens()) {
+            return;
+        }
+
+        long consensusTimestamp = recordItem.getConsensusTimestamp();
+        TransactionBody body = recordItem.getTransactionBody();
+        boolean isTokenDissociate = body.hasTokenDissociate();
+
+        recordItem.getRecord().getTokenTransferListsList().forEach(tokenTransferList -> {
+            TokenID tokenId = tokenTransferList.getToken();
+            EntityId entityTokenId = EntityId.of(tokenId);
+            EntityId payerAccountId = recordItem.getPayerAccountId();
+
+            insertFungibleTokenTransfers(
+                    consensusTimestamp, body, isTokenDissociate,
+                    tokenId, entityTokenId, payerAccountId,
+                    tokenTransferList.getTransfersList());
+
+            insertNonFungibleTokenTransfers(
+                    consensusTimestamp, body, tokenId, entityTokenId,
+                    payerAccountId, tokenTransferList.getNftTransfersList());
+        });
+    }
+
+    private void insertNonFungibleTokenTransfers(
+            long consensusTimestamp, TransactionBody body, TokenID tokenId,
+            EntityId entityTokenId, EntityId payerAccountId, List<com.hederahashgraph.api.proto.java.NftTransfer> nftTransfersList) {
+        for (NftTransfer nftTransfer : nftTransfersList) {
+            long serialNumber = nftTransfer.getSerialNumber();
+            if (serialNumber == NftTransferId.WILDCARD_SERIAL_NUMBER) {
+                // do not persist nft transfers with the wildcard serial number (-1) which signify an nft token
+                // treasury change
+                return;
+            }
+
+            EntityId receiverId = EntityId.of(nftTransfer.getReceiverAccountID());
+            EntityId senderId = EntityId.of(nftTransfer.getSenderAccountID());
+
+            com.hedera.mirror.common.domain.token.NftTransfer nftTransferDomain = new com.hedera.mirror.common.domain.token.NftTransfer();
+            nftTransferDomain.setId(new NftTransferId(consensusTimestamp, serialNumber, entityTokenId));
+            nftTransferDomain.setIsApproval(false);
+            nftTransferDomain.setReceiverAccountId(receiverId);
+            nftTransferDomain.setSenderAccountId(senderId);
+            nftTransferDomain.setPayerAccountId(payerAccountId);
+
+            com.hederahashgraph.api.proto.java.NftTransfer nftTransferInsideBody =
+                    findNftTransferInsideBody(nftTransfer, tokenId, body);
```

Comment:
Use var to hide the explicit import here and elsewhere.
```suggestion
            var nftTransferInsideBody = findNftTransferInsideBody(nftTransfer, tokenId, body);
```

Reply:
Done.